artichoke-backend/src/extn/core/regexp/regexp_test.rb

```diff
@@ -13,8 +14,24 @@ def regexp_initialize_already_init_literal
   r = /abc/in
   begin
     r.send(:initialize, 'xyz')
-    raise 'expected SecurityError'
-  rescue SecurityError => e
+    raise 'expected FrozenError'
+  rescue FrozenError => e
+    raise "got message: #{e.message}" unless e.message == "can't modify literal regexp"
+  end
+
+  r = /abc/in
+  begin
+    r.send(:initialize, '/xyz/')
+    raise 'expected FrozenError'
+  rescue FrozenError => e
+    raise "got message: #{e.message}" unless e.message == "can't modify literal regexp"
+  end
+
+  r = /abc/in
```

Comment:
I think for this test, it's the receiver that needs to not be a literal.

```suggestion
  r = Regexp.compile('abc')
```

Reply:
This test corresponds to the second case mentioned in #1985.

```bash
[3.1.2] > r.send(:initialize, Regexp.compile('abc'))
(irb):3:in `initialize': can't modify frozen Regexp: /abc/ (FrozenError)
        from (irb):3:in `<main>'
        from /usr/local/var/rbenv/versions/3.1.2/lib/ruby/gems/3.1.0/gems/irb-1.4.1/exe/irb:11:in `<top (required)>'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `load'
        from /usr/local/var/rbenv/versions/3.1.2/bin/irb:25:in `<main>'
```